fedimint-cli/src/client.rs

```diff
@@ -425,11 +425,15 @@ pub async fn handle_command(
                 // the amount we are withdrawing
                 BitcoinAmountOrAll::All => {
                     let balance =
-                        bitcoin::Amount::from_sat(client.get_balance().await.msats * 1000);
+                        bitcoin::Amount::from_sat(client.get_balance().await.msats / 1000);
                     let fees = wallet_module
                         .get_withdraw_fees(address.clone(), balance)
                         .await?;
-                    (balance - fees.amount(), fees)
+                    let amount = balance.checked_sub(fees.amount());
+                    if amount.is_none() {
+                        bail!("Not enough funds to pay fees");
+                    }
+                    (amount.unwrap(), fees)
```

Comment:
is balance is equal to fees this would result in `Some(0)` and pass. Probably would be better if it was instead

```rust 
let amount = balance.saturating_sub(fees.amount());
if amount == 0 {
	bail!("Not enough funds to pay fees");
}
(amount, fees)
```

Reply:
Good point. This case should throw an error when trying to broadcast the transaction, though. Somewhere we should also be checking that we aren't creating dust and IIRC that is currently handled by whatever bitcoin node we're using to broadcast. But don't recall.

Reply:
yeah, maybe just doing is enough

```
if amount <= 1_000 {
	bail!("Not enough funds to pay fees");
}
```

Reply:
Let's optimize this in a follow-up, there are a few cases to consider here (fees, dust, …). Currently the tx will fail+refund, which seems ok (just a bit slower feedback).